eea/dexterity/indicators/contentrules/version.py

```diff
@@ -128,9 +150,26 @@ def __call__(self):
 
         notify(ObjectClonedEvent(obj))
 
-        pr = getToolByName(obj, 'portal_repository')
+        pr = getToolByName(obj, "portal_repository")
         pr.save(obj=obj, comment=change_note)
 
+        # CHANGE URL OF FIGURES TO THE NEW DRAFT VERSION
+        for block_data in visit_blocks(obj, obj.blocks):
+            if (block_data.get("@type") == "embed_content" and
+                    "url" in block_data):
+                new_block = copy.deepcopy(block_data)
```

Comment:
I don't think this is necessary. In place modifying `new_block["url"] = url` should be enough. You can test this by restarting Plone instance n order to make sure there is no cache keeping the new value.